solutions/ceoi18-clo.mdx

```diff
@@ -0,0 +1,185 @@
+			<td>Profit change</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2200</td>
+			<td>-700</td>
+		</tr>
+		<tr>
+			<td>2</td>
+			<td>1800</td>
+			<td>-10</td>
+		</tr>
+		<tr>
+			<td>20</td>
+			<td>2550</td>
+			<td>-9999</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2000</td>
+			<td>-750</td>
+		</tr>
+		<tr>
+			<td>-1</td>
+			<td>1500</td>
+			<td>300</td>
+		</tr>
+		<tr>
+			<td>-6</td>
+			<td>1900</td>
+			<td>1500</td>
+		</tr>
+		<tr>
+			<td>-3</td>
+			<td>2400</td>
+			<td>4550</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Now we sort the array by clock rate in reverse, resulting in this:
+
+<center>
+<table className="text-center">
+	<tbody>
+		<tr>
+			<td>Core change</td>
+			<td>Clock rate</td>
+			<td>Profit change</td>
+		</tr>
+		<tr>
+			<td>20</td>
+			<td>2550</td>
+			<td>-9999</td>
+		</tr>
+		<tr>
+			<td>-3</td>
+			<td>2400</td>
+			<td>4550</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2200</td>
+			<td>-700</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2000</td>
+			<td>-750</td>
+		</tr>
+		<tr>
+			<td>-6</td>
+			<td>1900</td>
+			<td>1500</td>
+		</tr>
+		<tr>
+			<td>2</td>
+			<td>1800</td>
+			<td>-10</td>
+		</tr>
+		<tr>
+			<td>-1</td>
+			<td>1500</td>
+			<td>300</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Notice that in this order, as long as a customer order comes after a computer purchase, the computer will be able to satisfy the order as long as there are enough cores.
+
+This drastically simplifies the problem. Now we can simply define $\texttt{max_profits}[t][c]$ as the maximum profit we can gain from the first $t$ transactions given that we have $c$ cores left.
+Given this, the rest of the problem is simple knapsack.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+struct Transaction {
+	int cores;
+	int rate;
+	int price;
+};
+
+int main() {
+	vector<Transaction> poss_transactions;
+	int max_computers = 0;
+	int comp_num;
+	std::cin >> comp_num;
+	for (int c = 0; c < comp_num; c++) {
+		Transaction trans;
+		std::cin >> trans.cores >> trans.rate >> trans.price;
+		trans.price = -trans.price;
+		poss_transactions.push_back(trans);
+		max_computers += trans.cores;
+	}
+	int order_num;
+	std::cin >> order_num;
+	for (int o = 0; o < order_num; o++) {
+		Transaction trans;
+		std::cin >> trans.cores >> trans.rate >> trans.price;
+		trans.cores = -trans.cores;
+		poss_transactions.push_back(trans);
+	}
+	/*
+	 * if we sort like this, then the entire clock rate issue
+	 * goes away as long as we process them in order
+	 */
+	std::sort(
+		poss_transactions.begin(), poss_transactions.end(),
+		[] (const Transaction& a, const Transaction& b) -> bool {
+			return a.rate != b.rate ? a.rate > b.rate : a.price < b.price;
+		}
+	);
+
+	/*
+	 * max_profits[t][c] = the maximum profit we can gain from the first
+	 * t transactions given that we have $c$ cores left
```

Comment:
is there a reason why t isn't texified but c is

Reply:
oh
oh wait
what why did i even put that there